vivisect/symboliks/common.py

```diff
@@ -249,15 +249,19 @@ def setSymKid(self, idx, kid):
             todo = collections.OrderedDict({p._sym_id: p for p in oldkid.parents})
             done = set()
             while todo:
-                pid, parent = todo.popitem()
+                pid, parent = todo.popitem(last=False)
                 if pid in done:
                     continue
 
                 # track the objects whose cache has been cleared
                 done.add(pid)
                 parent.cache.clear()
                 # grow our todo list
-                todo.update({p._sym_id: p for p in parent.parents})
+                for prnt in parent.parents:
```

Comment:
please detail out what's going on.  this code is ripe for comments :)

Reply:
Aye. See docstring above.